File: src/gametest/scala/dev/krysztal/casualtiesbelow/gametest/PhysiologyGameTests.java
package dev.krysztal.casualtiesbelow.gametest;

import net.fabricmc.fabric.api.gametest.v1.GameTest;
import net.minecraft.gametest.framework.GameTestHelper;

/** Java discovery bridge for the Scala opioid GameTest scenarios. */
public final class PhysiologyGameTests {
    @GameTest(maxTicks = 400)
    public void analgesiaShockSuppression(GameTestHelper helper) {
        OpioidPhaseOneScenarios.analgesiaShockSuppression(helper);
    }

    @GameTest(maxTicks = 1000)
    public void overdoseDeath(GameTestHelper helper) {
        OpioidPhaseOneScenarios.overdoseDeath(helper);
    }

    @GameTest(maxTicks = 600)
    public void withdrawalTrio(GameTestHelper helper) {
        OpioidPhaseOneScenarios.withdrawalTrio(helper);
    }

    @GameTest(maxTicks = 20)
    public void syringeInjectsStoredOpioidDose(GameTestHelper helper) {
        SyringeScenarios.injectsStoredOpioidDose(helper);
    }

    @GameTest(maxTicks = 20)
    public void syringePartialInjectionKeepsRemainder(GameTestHelper helper) {
        SyringeScenarios.partialInjectionKeepsProportionalRemainder(helper);
    }

    @GameTest(maxTicks = 20)
    public void syringeInjectionPainLandsInOppositeArm(GameTestHelper helper) {
        SyringeScenarios.injectionPainLandsInOppositeArm(helper);
    }

    @GameTest(maxTicks = 20)
    public void syringeBatchWithStaleIdentityIsIgnored(GameTestHelper helper) {
        SyringeScenarios.batchWithStaleIdentityIsIgnored(helper);
    }

    @GameTest(maxTicks = 20)
    public void syringeCreativeInjectionNeverConsumes(GameTestHelper helper) {
        SyringeScenarios.creativeInjectionNeverConsumes(helper);
    }

    @GameTest(maxTicks = 20)
    public void infusionCauldronExposesUnfilteredFluidStorage(GameTestHelper helper) {
        PoppyFluidScenarios.infusionCauldronExposesUnfilteredStorage(helper);
    }

    @GameTest(maxTicks = 20)
    public void bucketConvertsWithInfusionCauldron(GameTestHelper helper) {
        PoppyFluidScenarios.bucketConvertsWithInfusionCauldron(helper);
    }

    @GameTest(maxTicks = 20)
    public void flowingFluidInvokerMapsLegacyLevels(GameTestHelper helper) {
        PoppyFluidScenarios.flowingFluidInvokerMapsLegacyLevels(helper);
    }

    @GameTest(maxTicks = 20)
    public void waterFluidInvokerDelegatesDropLogic(GameTestHelper helper) {
        PoppyFluidScenarios.waterFluidInvokerDelegatesDropLogic(helper);
    }
}

File: src/gametest/scala/dev/krysztal/casualtiesbelow/gametest/PhysiologyGameTests.scala
package dev.krysztal.casualtiesbelow.gametest

import net.minecraft.gametest.framework.GameTestHelper

import net.fabricmc.fabric.api.gametest.v1.GameTest

/** GameTest discovery entrypoint for the opioid, syringe and poppy-fluid scenarios. */
final class PhysiologyGameTests {

  @GameTest(maxTicks = 400)
  def opioidAcceleratesPainDecay(helper: GameTestHelper): Unit = {
    OpioidPhaseOneScenarios.opioidAcceleratesPainDecay(helper)
  }

  @GameTest(maxTicks = 1000)
  def overdoseDeath(helper: GameTestHelper): Unit = {
    OpioidPhaseOneScenarios.overdoseDeath(helper)
  }

  @GameTest(maxTicks = 600)
  def withdrawalTrio(helper: GameTestHelper): Unit = {
    OpioidPhaseOneScenarios.withdrawalTrio(helper)
  }

  @GameTest(maxTicks = 20)
  def syringeInjectsStoredOpioidDose(helper: GameTestHelper): Unit = {
    SyringeScenarios.injectsStoredOpioidDose(helper)
  }

  @GameTest(maxTicks = 20)
  def syringePartialInjectionKeepsRemainder(helper: GameTestHelper): Unit = {
    SyringeScenarios.partialInjectionKeepsProportionalRemainder(helper)
  }

  @GameTest(maxTicks = 20)
  def syringeInjectionPainLandsInOppositeArm(helper: GameTestHelper): Unit = {
    SyringeScenarios.injectionPainLandsInOppositeArm(helper)
  }

  @GameTest(maxTicks = 20)
  def syringeBatchWithStaleIdentityIsIgnored(helper: GameTestHelper): Unit = {
    SyringeScenarios.batchWithStaleIdentityIsIgnored(helper)
  }

  @GameTest(maxTicks = 20)
  def syringeCreativeInjectionNeverConsumes(helper: GameTestHelper): Unit = {
    SyringeScenarios.creativeInjectionNeverConsumes(helper)
  }

  @GameTest(maxTicks = 20)
  def infusionCauldronExposesUnfilteredFluidStorage(helper: GameTestHelper): Unit = {
    PoppyFluidScenarios.infusionCauldronExposesUnfilteredStorage(helper)
  }

  @GameTest(maxTicks = 20)
  def bucketConvertsWithInfusionCauldron(helper: GameTestHelper): Unit = {
    PoppyFluidScenarios.bucketConvertsWithInfusionCauldron(helper)
  }

  @GameTest(maxTicks = 20)
  def flowingFluidInvokerMapsLegacyLevels(helper: GameTestHelper): Unit = {
    PoppyFluidScenarios.flowingFluidInvokerMapsLegacyLevels(helper)
  }

  @GameTest(maxTicks = 20)
  def waterFluidInvokerDelegatesDropLogic(helper: GameTestHelper): Unit = {
    PoppyFluidScenarios.waterFluidInvokerDelegatesDropLogic(helper)
  }
}
